jedi/parser/tree.py

```diff
@@ -52,6 +53,19 @@ def is_node(node, *symbol_names):
         return type in symbol_names
 
 
+def _fix_forward_reference(annotation):
+    if isinstance(annotation, String):
```

Comment:
This is wrong in my opinion: It's not in the syntax tree where Python "hotfixes" the annotations. `py__annotations` should return a string and then the implementation should deal with that. (Parse the string (which is a `CompiledObject` by then) and evaluate that statement.

I can also prove your implementation wrong. Assume that the annotation is `type_factory(list)`. You would now create a name that includes brackets. Doesn't make a lot of sense. I guess adding a test case for this would also be nice. But if you don't want to, I will do it.